pkg/runtime/framework/plugins/mpi/mpi.go

```diff
@@ -103,7 +103,14 @@ func (m *MPI) EnforceMLPolicy(info *runtime.Info, trainJob *trainer.TrainJob) er
 
 	// TrainJob contains the actual information for the Trainer.
 	if trainJob.Spec.Trainer != nil && trainJob.Spec.Trainer.NumNodes != nil {
-		info.FindPodSetByName(constants.JobTrainerNode).Count = trainJob.Spec.Trainer.NumNodes
+		if node := info.FindPodSetByName(constants.JobNode); node != nil && node.Count != nil {
+			if ptr.Deref(info.RuntimePolicy.MLPolicySource.MPI.RunLauncherAsNode, false) {
+				// When runLauncherAsNode is enabled, 1 nodes should be allocated to launcher.
+				*node.Count = max(*trainJob.Spec.Trainer.NumNodes-1, 1)
```

Comment:
We might need to add validation in the future that if NumNodes >= 2 and RunLauncherAsNode = true, the `node` and `launcher` ReplicatedJob must be presented in the runtime.
@tenzen-y We can do it in the followup PRs, please can you add TODO statement ?

Reply:
Sure

Reply:
Done: https://github.com/kubeflow/trainer/issues/2550